sqlmesh/core/model/definition.py

```diff
@@ -814,7 +816,26 @@ def render(
         **kwargs: t.Any,
     ) -> t.Generator[QueryOrDF, None, None]:
         self._ensure_hydrated()
-        yield from self.seed.read(batch_size=self.kind.batch_size)
+
+        date_or_time_columns = []
+        bool_columns = []
+        string_columns = []
+        for name, tpe in (self.columns_to_types_ or {}).items():
+            if tpe in DATE_TIME_TYPES:
+                date_or_time_columns.append(name)
+            elif tpe == exp.DataType.build("boolean"):
+                bool_columns.append(name)
+            elif tpe in STRING_TYPES:
+                string_columns.append(name)
+
+        for df in self.seed.read(batch_size=self.kind.batch_size):
+            for column in date_or_time_columns:
+                df[column] = pd.to_datetime(df[column])
```

Comment:
this will only create date times right, what if you want to distinguish between date and datetime?

Reply:
All time-like types (date, timestamp, etc.) are convertible from the pandas datetime type. Tested this behavior with PyArrow.